### PR TITLE
pmapi basic

### PR DESCRIPTION
**Pull request by shamasis**
- all events are now namespaced as `execution.*[.*]` format
- added "Execution ID" specific events for error and assertions. Will help better tracking when async scripts come in.
- added `pm` in sandbox
- added `"use sandbox2";` script marker (which does not add legacy stuff)

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -70,9 +71,13 @@ _.assign(PostmanSandbox.prototype, {
             return callback(new Error('sandbox: no target provided for execution'));
         }
 
-        var id = uuid(),
-            executionEventName = `execution.${id}`,
-            executionTimeout = _.get(options, 'timeout', this.executionTimeout);
+        var id = _.isString(options.id) ? options.id : uuid(),
+            executionEventName = `execution.result.${id}`,
+            executionTimeout = _.get(options, 'timeout', this.executionTimeout),
+            cursor = _.get(options, 'cursor', {});
```
Comment (czardoz) [inline lib/index.js]: Since this is being mutated below with the execution ID, should we clone this?

File: lib/sandbox/execute.js [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
     StubTimers = require('./stub-timers'),
-    legacy = require('./postman-legacy-interface');
+    legacy = require('./postman-legacy-interface'),
+
+    EXECUTION_RESULT_EVENT_BASE = 'execution.result.',
+    EXECUTION_ERROR_EVENT = 'execution.error',
+    EXECUTION_ERROR_EVENT_BASE = 'execution.error.',
+
+    NONLEGACY_SANDBOX_MARKERS = {
+        '"use sandbox2";': true,
```
Comment (czardoz) [inline lib/sandbox/execute.js]: ... :elephant:

File: lib/index.js (hunk 1 above)
Comment (shamasis) [inline lib/index.js]: done

File: lib/sandbox/execute.js (hunk 2 above)
Comment (shamasis) [inline lib/sandbox/execute.js]: yeah... so that the uber OCD-ers can feel happy.

**Pull request merged**